benchmark/Snakefile

```diff
@@ -108,33 +108,44 @@ rule simulation:
         ts.dump(output[0])
 
 def generate_fs_from_ts(ts, sample_sets=None):
-
-    def allele_counts(ts, sample_sets=None):
-        if sample_sets is None:
-            sample_sets = [ts.samples()]
+    if sample_sets is None:
+        sample_sets = [ts.samples()]
+    def allele_counts(ts, sample_sets):
         def f(x):
             return x
         return ts.sample_count_stat(sample_sets, f, len(sample_sets),
                                     span_normalise=False, windows='sites',
                                     polarised=True, mode='site', strict=False)
 
-    mut_type = np.zeros(ts.num_sites)
+    # Mapping mutation type IDs to class of mutation (e.g., neutral, non-neutral)
+    mut_types = {}
+    for dfe in ts.metadata["stdpopsim"]["DFEs"]:
+        for mt in dfe["mutation_types"]:
+            mid = mt["slim_mutation_type_id"]
+            if not mid in mut_types:
+                mut_types[mid] = "neutral" if mt["is_neutral"] else "non_neutral"
+
+    print("These are the mutation types", flush=True)
+    print(mut_types, flush=True)
+
+    site_class = np.empty(ts.num_sites, dtype=object)
     for j, s in enumerate(ts.sites()):
         mt = []
         for m in s.mutations:
             for md in m.metadata["mutation_list"]:
                 mt.append(md["mutation_type"])
-        if len(set(mt)) > 1:
-            mut_type[j] = 3
-        else:
-            mut_type[j] = mt[0]
-
-    freqs = allele_counts(ts, [sample_sets])
+        site_class[j] = mut_types[mt[0]] if len(mt) == 1 else "double_hit" 
+    assert sum(site_class == None) == 0
+    print("Number of sites per class is:", flush=True)
+    unique, counts = numpy.unique(site_class, return_counts=True)
+    print(dict(zip(unique, counts)))
```

Comment:
so i think this code should be pushed to its own module in analysis, perhaps called `SFS_tools.py` and then imported to this workflow